### ISSUE TITLE
Make fontdiff compare renderings even if they aren't the same size

### ISSUE THREAD

**Issue by rsheeter** [googlei18n/nototools#162]
https://github.com/googlei18n/nototools/issues/158#issuecomment-175747707 suggests comparison is abandoned if the renders aren't the same size. For things like a ligature that has dropped part of it's contents ("fi" now renders just "i") that isn't ideal.

Resize the image to be able to provide visual comparison. Maybe just add whitespace?

**Comment by davelab6** [comment]
I am unable to get `convert` (with crop, resize, sample or scale) to work:

```
convert tmp_b.png  -adaptive-resize 93x157+0+0 tmp_b1.png;
compare -metric AE -verbose -size 93x157 tmp_a.png tmp_b1.png tmp_cmp.png;
tmp_a.png PNG 93x157 93x157+0+0 8-bit Grayscale Gray 256c 849B 0.000u 0:00.000
tmp_b1.png PNG 92x157 92x157+0+0 8-bit Grayscale Gray 256c 1.09KB 0.000u 0:00.000
compare: image widths or heights differ `tmp_a.png' @ error/compare.c/CompareImageCommand/990.
```

**Comment by dougfelt** [comment]
aligning the images would be a problem, I expect.

try something like convert -gravity center -background none -extent <width>x<height> src dst

**Comment by davelab6** [comment]
:100: 

`tmp_a`"

![tmp_a](https://cloud.githubusercontent.com/assets/261579/12628991/cc620a70-c513-11e5-9f74-fa5fd37c5f70.png)

`tmp_b`:

![tmp_b](https://cloud.githubusercontent.com/assets/261579/12628997/d22bbcb2-c513-11e5-8280-3e96a43607a1.png)

b via `convert -gravity center -background none -extent 93x157 tmp_b.png b.png;`

![b](https://cloud.githubusercontent.com/assets/261579/12629018/e413feee-c513-11e5-873c-7acb18d9b116.png)

Then `tmp_cmp.png` via

    compare -metric AE -verbose -size 93x157 tmp_a.png b.png tmp_cmp.png;
        tmp_a.png PNG 93x157 93x157+0+0 8-bit Grayscale Gray 256c 849B 0.000u 0:00.000
        b.png PNG 93x157 93x157+0+0 8-bit Grayscale Gray 256c 1.02KB 0.000u 0:00.000
        Image: tmp_a.png
          Channel distortion: AE
            gray: 1583
            all: 1583
        tmp_a.png=>tmp_cmp.png PNG 93x157 93x157+0+0 8-bit sRGB 168c 1.58KB 0.000u 0:00.000

![tmp_cmp](https://cloud.githubusercontent.com/assets/261579/12629045/fc18955e-c513-11e5-812c-e49d7b4ec15c.png)

**Comment by dougfelt** [comment]
Yes, the alignment issue I cautioned about.  At least you can run compare now.

**Comment by davelab6** [comment]
This is actually working fine, I think, because the glyph did move down in this change :)

http://stackoverflow.com/a/33673440/1174651 offers this code as an alternate to using the `compare` tool itself,

    convert '(' tmp_a.png -grayscale Rec709Luminance ')' \
        '(' b.png -grayscale Rec709Luminance ')' \
        '(' -clone 0-1 -compose darken -composite ')' \
        -channel RGB -combine diff.png

which produces this beauty:

![diff](https://cloud.githubusercontent.com/assets/261579/12630056/8c7f76c2-c518-11e5-97b4-64c200e93d2f.png)

**Comment by davelab6** [comment]
> I am unable to get `convert` (with crop, resize, sample or scale) to work:
>
>    convert tmp_b.png  -adaptive-resize 93x157+0+0 tmp_b1.png;
>    identify tmp_b1.png ;
>    tmp_b1.png PNG 92x157 92x157+0+0 8-bit Grayscale Gray 256c 1.09KB 0.000u 0:00.000

Ah, appending an exclamation mark to the extent forces image scaling without preserving ratio, 

    convert tmp_b.png  -adaptive-resize 93x157+0+0! tmp_b1.png;
    identify tmp_b1.png 
    tmp_b1.png PNG 93x157 93x157+0+0 8-bit sRGB 256c 1.14KB 0.000u 0:00.000

then compare works

**Comment by davelab6** [comment]
Note for me to read later
* http://www.imagemagick.org/Usage/compare/
* http://www.imagemagick.org/script/compare.php

**Comment by jamesgk** [comment]
I think we'd probably want to just resize the image with extent, rather than scaling it which I think is what forced resize does.

Using `-background none` actually makes the background non-white for comparison purposes (the same effect as e.g. `-background black`. While that may not have been intended, it may be helpful for catching errors such as "fi" turning to "i" (since glyphs with very different widths will have very large diff values).

**Comment by davelab6** [comment]
Yes, I think that is a feature not a bug for that error catching reason :)

**Comment by jamesgk** [comment]
OK. I can submit a PR for this once #163 is merged. It seems like it would be useful to actually save the diff images somewhere (maybe sort them by difference as well)? That can be a separate issue too.
